Content.Server/Chemistry/Components/FoamSolutionAreaEffectComponent.cs

```diff
@@ -40,20 +38,22 @@ protected override void ReactWithEntity(EntityUid entity, double solutionFractio
             if (!_entMan.TryGetComponent(entity, out BloodstreamComponent? bloodstream))
                 return;
 
+            var invSystem = EntitySystem.Get<InventorySystem>();
+
             // TODO: Add a permeability property to clothing
             // For now it just adds to protection for each clothing equipped
             var protection = 0f;
-            if (_entMan.TryGetComponent(entity, out InventoryComponent? inventory))
+            if (invSystem.TryGetSlots(entity, out var slotDefinitions))
             {
-                foreach (var slot in inventory.Slots)
+                foreach (var slot in slotDefinitions)
                 {
-                    if (slot == EquipmentSlotDefines.Slots.BACKPACK ||
-                        slot == EquipmentSlotDefines.Slots.POCKET1 ||
-                        slot == EquipmentSlotDefines.Slots.POCKET2 ||
-                        slot == EquipmentSlotDefines.Slots.IDCARD)
+                    if (slot.Name == "back" ||
+                        slot.Name == "pocket1" ||
+                        slot.Name == "pocket2" ||
+                        slot.Name == "id0")
```

Comment:
I think this is just supposed to be "id" not "id0"

Reply:
yep, a typo. i dont really like using strings as slot-names but it was the best i could come up with... maybe we'll circle back to the slotenum afterall. idunno. any ideas?

Reply:
enums might be better just because it would prevent these sorta typos from introducing bugs, but I'm not really sure.

The way that prototype-id strings are also sometimes hard coded already bothers me. If they're yaml data-fields, at least there's the YAML linter. But, as an example, there's no guarantee that the "puddle" wasn't misspelled somewhere in the puddle system when spawning in a new puddle.